ios/RNTypeValidation.m

```diff
@@ -0,0 +1,211 @@
+//
+//  RNTypeValidation.m
+//  RNJudo
+//
+//  Copyright (c) 2020 Alternative Payments Ltd
+//
+//  Permission is hereby granted, free of charge, to any person obtaining a copy
+//  of this software and associated documentation files (the "Software"), to deal
+//  in the Software without restriction, including without limitation the rights
+//  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+//  copies of the Software, and to permit persons to whom the Software is
+//  furnished to do so, subject to the following conditions:
+//
+//  The above copyright notice and this permission notice shall be included in all
+//  copies or substantial portions of the Software.
+//
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+//  SOFTWARE.
+
+#import "RNTypeValidation.h"
+
+@implementation RNTypeValidation
+//----------------------------------------------
+// MARK: - Configs Validation
+//----------------------------------------------
+
++ (nullable NSString *)isConfigurationValid:(NSDictionary *) properties {
+    NSDictionary *configuration = properties[@"configuration"];
+    
+    if(![configuration[@"judoId"] isKindOfClass:[NSString class]]) {
+        return @"judoId property is not valid, judoId should be a string.";
+    }
+    
+    if (![configuration[@"siteId"] isKindOfClass:[NSString class]] && configuration[@"siteId"] != nil ) {
+        return @"siteId is not valid, siteId should be a string";
+    }
+    
+    if (![self isAmountDictionaryValid: configuration[@"amount"]]) {
+        return @"amount is not valid, amount should be a dictionary of type JPAmount";
+    }
+    
+    if (![self isReferenceValid:configuration[@"reference"]]) {
+        return @"reference is not valid, reference should be a dictionary of type JPReference";
```

Comment:
We should validate the `consumerReference` type and nullability as well, as well as the optional `paymentReference` and `metadata` props